jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -682,6 +666,153 @@ ecma_string_to_utf8_string (const ecma_string_t *string_desc_p, /**< ecma-string
 } /* ecma_string_to_utf8_string */
 
 /**
+ * Lengths for numeric string values
+ */
+static const uint32_t nums_with_ascending_length[] =
+{
+  1u,
+  10u,
+  100u,
+  1000u,
+  10000u,
+  100000u,
+  1000000u,
+  10000000u,
+  100000000u,
+  1000000000u
+};
+
+/**
+ * Maximum length of numeric strings
+ */
+static const uint32_t max_uint32_len = (uint32_t) (sizeof (nums_with_ascending_length) / sizeof (uint32_t));
+
+/**
+ * Get size of the number stored locally in the string's descriptor
+ *
+ * Note: the represented number size and length are equal
+ *
+ * @return size in bytes
+ */
+static ecma_length_t __attr_always_inline___
+ecma_string_get_number_in_desc_size (const uint32_t uint32_number) /**< number in the string-descriptor */
+{
+  ecma_length_t size = 1;
+
+  while (size < max_uint32_len && uint32_number >= nums_with_ascending_length[size])
+  {
+    size++;
+  }
+  return size;
+} /* ecma_string_get_number_in_desc_size */
+
+/**
+ * Get size of container heap number of ecma-string
+ *
+ * Note: the number size and length are equal
+ *
+ * @return number of bytes in the buffer
+ */
+static lit_utf8_size_t __attr_always_inline___
+ecma_string_get_heap_number_size (mem_cpointer_t number_cp) /**< Compressed pointer to an ecma_number_t */
+{
+  const ecma_number_t *num_p = ECMA_GET_NON_NULL_POINTER (ecma_number_t, number_cp);
+  lit_utf8_byte_t buffer[ECMA_MAX_CHARS_IN_STRINGIFIED_NUMBER];
+
+  return ecma_number_to_utf8_string (*num_p, buffer, sizeof (buffer));
+} /* ecma_string_get_heap_number_size */
```

Comment:
Yes, but the function belongs to that group I placed it. I like when function implementation follows the declaration order in the header, but if you insist, I can change it.